designdocs/delegation.md

```diff
@@ -22,175 +22,129 @@ The primary motivation for delegation is to support End Users of the DSNP platfo
-* Does not include a "block/ban" feature for delegation, which is under discussion; the belief is that a Provider also ought to be able to permanently refuse service to a given `MsaId`, which further supports the idea of a mutually agreed upon relationship.
+* Does not include a "block/ban" feature for delegation, which is under discussion; the belief is that a Provider also ought to be able to permanently refuse service to a given MSA id, which further supports the idea of a mutually agreed upon relationship.
 
 ## Proposal
-The proposed solution is to give End Users the ability to create an on-chain `MsaId` through an authorized provider. End Users can also transparently authorize and manage their own Providers and permissions, either directly using a native token or through an explicitly authorized Provider. Additionally, we allow `MsaIds` to be directly purchased using a native token.
+The proposed solution is to give End Users the ability to create an on-chain MSA id through an authorized provider. End Users can also transparently authorize and manage their own Providers and permissions, either directly using a native token or through an explicitly authorized Provider. Additionally, we allow MSA ids` to be directly purchased using a native token.
 
 ### API (extrinsics)
 * All names are placeholders and may be changed.
 * All extrinsics must emit an appropriate event with all parameters for the call, unless otherwise specified.
 * Errors in the extrinsics must have different, reasonably-named error enums for each type of error for ease of debugging.
-* "Owner only" means the caller must own the delegator`MsaId`.
+* "Owner only" means the caller must own the delegatorMSA id.
 * Events are not deposited for read-only extrinsic calls.
 
-#### create_account_with_provider
-Creates a new `MsaId` on behalf of an MSA and adds the origin as the MSA's provider.
+#### create_sponsored_account_with_delegation
+Creates a new MSA on behalf of a delegator and adds the origin held MSA as its provider.
+
   * Parameters:
-      1. `payload`:
-         1. `data` - this is what the MSA owner must sign and provide to the provider beforehand.
-             * `msa_id` - the provider, of type `MsaId`
-             * `permission` a value indicating the permission to be given to the provider
-         2. `public_key` - The authorizing key used to create `signature`
-         3. `signature` - The signature of the hash of `data`
-  * Event:  `IdentityCreated`, with the `public_key`, the newly created `MsaId`, and `msa_id`
-  * Restrictions:  origin must own `msa_id` in the payload.
-
-#### add_self_as_provider(payload)
-Adds the `MsaId` in the payload as a provider, to an MSA owning `delegator_msa_id`
+      1. `add_provider_payload` - this is what the holder of delegator_key must sign and provide to the provider beforehand.
+          * `authorized_msa_id` - the provider, of type `MessageSourceId`
```

Comment:
technically the `authorized_msa_id` should be part of the payload, but not part of the extrinsic as it can be looked up from the origin's key.

I'd be fine with permission being a separate param and then instead of a payload, it is just params that build the same payload as the user signed. We just need to have all that documentation about the payload the user should sign over on `proof`

Reply:
I modified the term `payload` in the document to match the parameter name of `add_provider_payload` on the extrinsic and intended to signify that `authorized_msa_id` and `permission` are constituents of `add_provider_payload`. Perhaps explaining how the payload is signed as proof would make it more clear.